agents_infra/agents_infra/agents/base.py

```diff
@@ -622,81 +558,120 @@ def post_data(
                             'POST failed after %d attempts' % max_retries
                         )
 
-    def fetch_command_from_controller(
-        self, suffix='command/fetch/', timeout=5, api_key=None, **kwargs
+    def get_data(
+        self,
+        url,
+        to_controller=True,
```

Comment:
For consistency, I think in `get_data`, we should rename `to_controller` into `from_controller` :)